pkg/connectors/graylog/connector.go

```diff
@@ -0,0 +1,149 @@
+package graylog
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	html "html/template"
+	"io"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/synyx/tuwat/pkg/connectors"
+	"github.com/synyx/tuwat/pkg/connectors/common"
+)
+
+type Connector struct {
+	config Config
+	client *http.Client
+}
+
+type Config struct {
+	Tag     string
+	Cluster string
+	common.HTTPConfig
+}
+
+func NewConnector(cfg *Config) *Connector {
+	c := &Connector{config: *cfg, client: cfg.HTTPConfig.Client()}
+
+	return c
+}
+
+func (c *Connector) Tag() string {
+	return c.config.Tag
+}
+
+func (c *Connector) Collect(ctx context.Context) ([]connectors.Alert, error) {
+	sourceAlerts, err := c.collectAlertEvents(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var alerts []connectors.Alert
+
+	for _, sourceAlert := range sourceAlerts.Events {
+		var streams []string
+		for _, stream := range sourceAlerts.Context.Streams {
+			streams = append(streams, stream.Title)
+		}
+
+		labels := map[string]string{
+			"Source": sourceAlert.Event.Source,
+			"Stream": strings.Join(streams, ","),
+		}
+		alert := connectors.Alert{
+			Labels:      labels,
+			Start:       parseTime(sourceAlert.Event.TimeRangeStart),
+			State:       connectors.Warning,
+			Description: sourceAlert.Event.Message,
+			Details:     sourceAlerts.Context.EventDefinitions[sourceAlert.Event.EventDefinitionId].Description,
+			Links: []html.HTML{
+				html.HTML("<a href=\"" + c.config.URL + "/alerts" + "\" target=\"_blank\" alt=\"Home\">🏠</a>"),
+			},
+		}
+		alerts = append(alerts, alert)
+	}
+
+	return alerts, nil
+}
+
+func (c *Connector) String() string {
+	return fmt.Sprintf("Alertmanager (%s)", c.config.URL)
```

Comment:
This confused me while running it locally, as there is no Alertmanager connector that I know of. Should this read `"Graylog (%s)"`?

Reply:
You're absolutely correct - a copy/paste error.